### PR TITLE
[Merged by Bors] - doc: add docstrings for vector and matrix notation

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/VecNotation.lean [hunk 1]
```diff
@@ -61,7 +61,14 @@ def vecCons {n : ℕ} (h : α) (t : Fin n → α) : Fin n.succ → α :=
...
 #align matrix.vec_cons Matrix.vecCons
 
-/-- Construct a vector `Fin n → α` using `Matrix.vecEmpty` and `Matrix.vecCons`. -/
+/-- `![...]` notation is used to construct a vector `Fin n → α` using `Matrix.vecEmpty` and
+`Matrix.vecCons`.
+
+For instance, `![a, b, c] : Fin 3` is syntax for `vecCons a (vecCons b (vecCons c vecEmpty))`.
+
+Note that this should not be used as syntax for `Matrix` as it generates a term with the wrong type.
+The `!![a, b; c, d]` syntax should be used instead (from `Mathlib/Data/Matrix/Notation`).
```
Comment (adomani) [inline Mathlib/Data/Fin/VecNotation.lean:70]:
```suggestion
The `!![a, b; c, d]` syntax should be used instead (from `Matrix.matrixNotation`).
```
Would using this create a link in the docs?  Or something similar?

**Comment by adomani** [comment]
Thanks a lot: I sure would have benefitted from the explicitness in the explanation!

File: Mathlib/Data/Fin/VecNotation.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Data/Fin/VecNotation.lean]: Done, with some rewording.

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Comment by eric-wieser** [comment]
bors merge

(I don't expect docstrings to fail)

**Comment by eric-wieser** [comment]
bors merge

File: Mathlib/Data/Matrix/Notation.lean [hunk 2]
```diff
@@ -72,9 +72,29 @@ end toExpr
...
+
+For instance:
+* `!![a, b, c; d, e, f]` is the matrix with two rows and three columns, of type
+  `Matrix (Fin 2) (Fin 3) α`
+* `!![a, b, c]` is a row vector of type `Matrix (Fin 1) (Fin 3) α` (see also `Matrix.row`).
+* `!![a; b; c]` is a column vector of type `Matrix (Fin 3) (Fin 1) α` (see also `Matrix.col`).
+
+This notation implements some special cases:
+
+* `![,,]`, with `n` `,`s, is a term of type `Matrix (Fin 0) (Fin n) α`
```
Comment (Julian) [inline Mathlib/Data/Matrix/Notation.lean:85]: Not at lean otherwise I'd check, but is this right? I.e. it's `![,,]` and friends, not `!![,,]`?

Comment (eric-wieser) [reply Mathlib/Data/Matrix/Notation.lean:85]: Whoops, this was a mistake!

**Pull request closed without merging**